### PR TITLE
[JENKINS-64816] prepare for support of node monitors by JCasC

### PR DESCRIPTION
**Pull request by mawinter69**
In order that configuration as code plugin can properly support node monitors we need to add databound constructors.

See [JENKINS-64816](https://issues.jenkins.io/browse/JENKINS-64816).

### Testing done
No functional changes, verified in the context of https://github.com/jenkinsci/configuration-as-code-plugin/pull/2392 that with this change the node monitors can then be configured via CasC

### Proposed changelog entries

- Prepare node monitors to work with configuration as code.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java
Comment (daniel-beck) [inline core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java:69]: Looks risky given line 95. Does this create a duplicate ExtensionList entry? Is this newly invocation order dependent?

**Review by daniel-beck — COMMENTED** [review]
Looks like we might be able to get rid of a few `#newInstance` as well, but reasonable to consider that out of scope.

File: core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java
Comment (mawinter69) [reply core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java:69]: ahh I missed that line 95. Fixed it hopefully.

File: core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java
Comment (daniel-beck) [inline core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java:101]: Probably reasonable to delete (after confirming nobody calls it to be safe). This method only exists to instantiate the class for `@Extension`, and based on https://github.com/jenkinsci/jenkins/commit/f592d33a7ecf2c5806ae41c827f5d93b17da87b2#diff-61ebef87257e01fc5f2b0a6c82d54c638748cb3e10e5f476170e03124356bc91R102-R106 only because it used to be conditionally instantiated, with https://github.com/jenkinsci/jenkins/commit/4825967fe322bd84bf968773d37e4655366120b9#diff-61ebef87257e01fc5f2b0a6c82d54c638748cb3e10e5f476170e03124356bc91R78 having been incomplete cleanup.

Comment (mawinter69) [reply core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java]: Seems the method is not used anywhere. The same method exists for TemporarySpaceMonitor, so lets delete them both

File: core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java
Comment (daniel-beck) [inline core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java:70]: Unsure whether we want `@since` for this. Technically, yes?

File: core/src/main/java/hudson/node_monitors/ResponseTimeMonitor.java
Comment (daniel-beck) [inline core/src/main/java/hudson/node_monitors/ResponseTimeMonitor.java:60]: Unsure whether we want `@since` for this. Technically, yes?

**Review by daniel-beck — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
nice

**Comment by NotMyFault** [comment]
Hey @mawinter69,

would you mind addressing or commenting on the outstanding concerns, please?

File: core/src/main/java/hudson/node_monitors/ResponseTimeMonitor.java
Comment (mawinter69) [reply core/src/main/java/hudson/node_monitors/ResponseTimeMonitor.java:60]: dito

File: core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java
Comment (mawinter69) [reply core/src/main/java/hudson/node_monitors/DiskSpaceMonitor.java:70]: I don't see really the need for this here.

**Comment by daniel-beck** [comment]
Noting that comments have been addressed by declining.

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
